docs/_docs/reference/other-new-features/export.md

```diff
@@ -123,13 +137,43 @@ Export clauses also fill a gap opened by the shift from package objects to top-l
 of internal compositions available to users of a package. Top-level definitions are not wrapped in a user-defined object, so they can't inherit anything. However, top-level definitions can be export clauses, which supports the facade design pattern in a safer and
 more flexible way.
 
+## Export Clauses in Extensions
+
+An export clause may also appear in an extension.
+
+Example:
+```scala
+class StringOps(x: String):
+  def *(n: Int): String = ...
+  def capitalize: String = ...
+
+extension (x: String)
+  def take(n: Int): String = x.substring(0, n)
+  def drop(n: Int): String = x.substring(n)
+  private def moreOps = new StringOps(x)
+  export moreOps.*
+```
+In this case the qualifier expression must be an identifier that refers to a unique parameterless extension method in the same extension clause. The export will create
+extension methods for all accessible term members
+in the result of the qualifier path. For instance, the extension above would be expanded to
+```scala
+extension (x: String)
+  def take(n: Int): String = x.substring(0, n)
+  def drop(n: Int): String = x.substring(n)
+  private def moreOps = StringOps(x)
```

Comment:
`StringOps` could be a value class. Even if it isn't I would guess that JVM escape analysis would remove the allocation in this instance, since forwarders like `moreOps` are small enough to be inlined.

Reply:
you can export from an object that already contains extensions as follows:
```scala
object StrSyntax {
  // these exports will not be in global scope
  extension (s: String) {
    def x2 = s * 2
    def %: [T](t: T) = s"$t %: $s"
  }
}

extension (s: String) {
  private def syntax = StrSyntax
  export syntax.*
}

@main def Test =
  println("Hello, world!".x2)
  println(23 %: "Hello, world!")
```

Reply:
Could we add a test for this example? It's nice for code reviewers to be able see there is no allocation so that object based example looks nice to me.

Reply:
Sorry, I tested my example and it actually doesn't work, see my comment https://github.com/lampepfl/dotty/pull/14497#pullrequestreview-887612148

Reply:
No, unification is neither specified nor implemented. I think it's not necessary to go that far to support the primary use case (which is replacing implicit classes that get their methods from some parent class(es)).

Reply:
also, you can export those methods from outside of the extension block and avoid the duplication of parameters

Reply:
perhaps export wildcard should not reexport extension methods from within an extension block

Reply:
filtering out of extensions from imports implemented in 61b1a7b7468e18e2a6b9d7fa0ddc1c3bb239d1b2